addons/mail/models/ir_attachment.py

```diff
@@ -38,15 +38,17 @@ def register_as_main_attachment(self, force=True):
             with contextlib.suppress(AccessError):
                 related_record.message_main_attachment_id = self
 
-    def _delete_and_notify(self):
+    def _delete_and_notify(self, message=None):
         for attachment in self:
             if attachment.res_model == 'mail.channel' and attachment.res_id:
                 target = self.env['mail.channel'].browse(attachment.res_id)
             else:
                 target = self.env.user.partner_id
-            self.env['bus.bus']._sendone(target, 'ir.attachment/delete', {
-                'id': attachment.id,
-            })
+            vals = {'id': attachment.id}
+            if message:
```

Comment:
Pay attention that you are doing a write in a loop, with cache mechanism this will probably lead to a single update but good practice would be to do it outside of the loop.